src/olympia/editors/tests/test_helpers.py

```diff
@@ -501,6 +509,49 @@ def test_nomination_to_public(self, sign_mock):
         self._check_score(amo.REVIEWED_ADDON_FULL)
 
     @patch('olympia.editors.helpers.sign_file')
+    def test_public_addon_with_version_awaiting_review_to_public(
```

Comment:
Amazingly, though it's covered by `test_views.py`, this scenario wasn't unit tested :(